mhw-cs-plugin-loader/Preloader.cpp

```diff
@@ -51,14 +49,33 @@ void open_console() {
     setvbuf(stdout, nullptr, _IOFBF, 1000);
 }
 
+// Returns pointer to the IMAGE_LOAD_CONFIG_DIRECTORY64.SecurityCookie value.
+uint64_t* get_security_cookie_pointer()
+{
+    auto image_base = (uint64_t)GetModuleHandle(NULL);
+    IMAGE_DOS_HEADER* dos_header = (IMAGE_DOS_HEADER*)image_base;
+    IMAGE_NT_HEADERS* nt_headers = (IMAGE_NT_HEADERS*)(image_base + dos_header->e_lfanew);
+    if (nt_headers->OptionalHeader.NumberOfRvaAndSizes >= IMAGE_DIRECTORY_ENTRY_LOAD_CONFIG)
+    {
+        auto load_config_directory = nt_headers->OptionalHeader.DataDirectory[IMAGE_DIRECTORY_ENTRY_LOAD_CONFIG];
+        if (load_config_directory.VirtualAddress != 0 && load_config_directory.Size != 0)
+        {
+            IMAGE_LOAD_CONFIG_DIRECTORY64* load_config = (IMAGE_LOAD_CONFIG_DIRECTORY64*)(image_base + load_config_directory.VirtualAddress);
+            return (uint64_t*)load_config->SecurityCookie;
+        }
+    }
+
+    return nullptr;
+}
```

Comment:
Code Style (Braces) 😆 Not very important, just thought I'd mention it, there's a couple inconsistencies